build/utils.go

```diff
@@ -67,11 +69,19 @@ func toBuildkitExtraHosts(inp []string, mobyDriver bool) (string, error) {
 		if !ok || host == "" || ip == "" {
 			return "", errors.Errorf("invalid host %s", h)
 		}
-		// Skip IP address validation for "host-gateway" string with moby driver
-		if !mobyDriver || ip != mobyHostGatewayName {
-			if net.ParseIP(ip) == nil {
-				return "", errors.Errorf("invalid host %s", h)
+		// If the IP Address is a "host-gateway", replace this value with the
+		// IP address provided by the worker's label.
+		if ip == mobyHostGatewayName {
+			hgip, err := nodeDriver.HostGatewayIP(ctx)
+			if err != nil {
+				return "", err
+			} else if hgip == nil {
+				return "", errors.Errorf("unable to derive the IP value for host-gateway")
 			}
+			ip = hgip.String()
+		}
+		if net.ParseIP(ip) == nil {
```

Comment:
_This one_ should actually be an `else if` (or at least, I would not expect `hgip.String()` to be producing an invalid IP; if that's the case, that would be a bug in Go's stdlib 😅)

 ```suggestion
		} else if net.ParseIP(ip) == nil {
```

Reply:
I think you missed this one 😅